internal/tasks/config_helpers.go

```diff
@@ -18,7 +18,18 @@ func ContainsParameter(b []byte) bool {
 	return paramsRegex.Match(b)
 }
 
-func ReplaceParameters(b []byte, objectMeta api.ObjectMeta) ([]byte, error) {
+func ValidateParameterFormat(b []byte) error {
+	matches := paramsRegex.FindAllStringSubmatch(string(b), -1)
```

Comment:
In general the validation is too loose.  It allows all kind of illegal formats, as long as one of the expected formats appears in the parenthasis.
`{{ abcd device.metadata.name zzz}}`
`{{ {{{{ device.metadata.labels[key]  @@@@@}}`
Taking a stricter regex can do the job.  Example:
`(?P<full>{{\s*(?:(?P<name>device\.metadata\.name)|(?P<label>device\.metadata\.labels\[(?P<key>.*)\]))\s*}})`